doc/source/whatsnew/v2.2.0.rst

```diff
@@ -189,7 +189,7 @@ Groupby/resample/rolling
 
 Reshaping
 ^^^^^^^^^
--
+- Bug in :func:`merge` not following documented sort behavior in certain cases (:issue:`54611`)
```

Comment:
Might be good to have this in the "notable bug fix" section since it fixes a lot of issues!

Reply:
Moved to the notable section